File: src/crowstream/reproduction/typeDefs.js
export const userVideoMetadataTypeDef = `
  type UserVideoMetadata {
      ID: ID!
      user_id: Int!
      video_id: Int!
      video_progress: Float!
      video_progress_time: String!
  }
  input UserVideoMetadataInput {
      user_id: Int!
      video_id: Int!
      video_progress: Float!
      video_progress_time: String!
  }`;

export const userVideoMetadataQueries = `
      userVideoMetadataById(id: Int!): UserVideoMetadata!
  `;

export const userVideoMetadataMutations = `
    createUserVideoMetadata(userVideoMetadata: UserVideoMetadataInput!): UserVideoMetadata!
`;

File: api_gateway/schema/reproduction/video_metadata/type_defs.py
"""reproduction Type Defs"""

# Utilities
import graphene

class UserVideoMetadata(graphene.ObjectType):
    ID = graphene.ID(name='ID')
    user_id = graphene.String(name='user_id')
    video_id = graphene.Int(name='video_id')
    video_progress = graphene.Float(name='video_progress')
    video_progress_time = graphene.String(name='video_progress_time')

class UserVideoMetadataInput(graphene.InputObjectType):
    user_id = graphene.String(name='user_id')
    video_id = graphene.Int(name='video_id')
    video_progress = graphene.Float(name='video_progress')
    video_progress_time = graphene.String(name='video_progress_time')